torch/autograd/gradcheck.py

```diff
@@ -160,18 +161,41 @@ def get_stride(size):
     return jacobian
 
 
-def get_analytical_jacobian(input, output, nondet_tol=0.0, grad_out=1.0):
-    # it is easier to call to_dense() on the sparse output than
-    # to modify analytical jacobian
-    if output.is_sparse:
-        raise ValueError('Sparse output is not supported at gradcheck yet. '
-                         'Please call to_dense() on the output of fn for gradcheck.')
-    if output.layout == torch._mkldnn:  # type: ignore
-        raise ValueError('MKLDNN output is not supported at gradcheck yet. '
-                         'Please call to_dense() on the output of fn for gradcheck.')
-    diff_input_list = list(iter_tensors(input, True))
-    jacobian = make_jacobian(input, output.numel())
-    jacobian_reentrant = make_jacobian(input, output.numel())
+def check_analytical_jacobian(inputs, output, nondet_tol, grad_out, check_batched_grad,
+                              check_grad_dtypes, raise_exception):
```

Comment:
nit: If we're refactoring this anyways, it would be nice to add type hints (either now or in the future). For example, it's important to note here that inputs is a `Tuple` (I think? At least from reading the original code this seemed to be the case)

Reply:
mypy has been a little tricky, lets save this for the future